File: lessons/gen_scripts/perl/scripts/e41.branch.pl
#!/usr/bin/env perl -w
# Switch.pm - a source-filter module providing Perl6-style switch/case -
#  isn't core anymore and needs installing yourself (see ../README.md's
#  "Perl Modules (CPAN)" section). It's shown here for how a genuine
#  switch/case reads in Perl; e40.branch.pl's hash dispatch is the
#  dependency-free technique actually recommended today.
use Switch;

my $menu = <<'END';
Select an item from the menu.

  1 - Coffee
  2 - Espresso
  3 - Latte
  4 - Machiato
  5 - Capucino
  6 - Mocha
  7 - Tea

Make your selection: 
END
chomp $menu; # strip the heredoc's own trailing newline - no linebreak
             #  before the answer is typed
print $menu;
my $selection = <>; chomp $selection;

switch ($selection) {
    case 1 { print "You selected a Coffee\n" }
    case 2 { print "You selected an Espresso\n" }
    case 3 { print "You selected a Latte\n" }
    case 4 { print "You selected a Machiato\n" }
    case 5 { print "You selected a Capucino\n" }
    case 6 { print "You selected a Mocha\n" }
    case 7 { print "You selected a Tea\n" }
    else   { print "You have not entered a valid selection\n" }
}

File: lessons/gen_scripts/python2/scripts/e41.branch.py
#!/usr/bin/env python2
import sys  # system library for standard input and output

# build a menu string to output to user
menu = """\
Select an item from the menu.

  1 - Coffee
  2 - Espresso
  3 - Latte
  4 - Machiato
  5 - Capucino
  6 - Mocha
  7 - Tea

Make your selection: """

sys.stdout.write(menu)              # output menu and prompt
selection = int(sys.stdin.read(1))  # read one number

# python 2 has no switch/match statement (match/case is python 3.10+,
#  see python3's e40 lesson) - a dict keyed by value is the idiomatic,
#  dependency-free stand-in for a multiway branch, same as e40.branch.pl
options = {
  1: "You selected a Coffee",
  2: "You selected an Espresso",
  3: "You selected a Latte",
  4: "You selected a Machiato",
  5: "You selected a Capucino",
  6: "You selected a Mocha",
  7: "You selected a Tea",
}

print options.get(selection, "You have not entered a valid selection")
